telethon/telegram_client.py

```diff
@@ -104,8 +106,9 @@ def __init__(self, session, api_id, api_hash,
         """Sends a code request to the specified phone number"""
         phone = EntityDatabase.parse_phone(phone) or self._phone
         result = await self(SendCodeRequest(phone, self.api_id, self.api_hash))
-        self._phone = phone
-        self._phone_code_hash = result.phone_code_hash
+        if result:
```

Comment:
I really aren't sure about this. I think not crashing with a `None` result here is worse than crashing, because it will surely fail without the user noticing.